### PR TITLE
Simplify version definition

### PR DESCRIPTION
**Pull request by Axel-Naumann**
This makes releasing waaay simpler, more reliable, and possibly at some point convertible to a Github action?

### CODE REVIEW

**Review by gganis — APPROVED** [review]
Looks good to me. Thanks. G

**Comment by pcanal** [comment]
> and- in its current implementation - a working ROOT build as part of the version update.

Why is/was that?

File: core/foundation/inc/RVersion.h [hunk 1]
```diff
@@ -1,23 +1,8 @@
...
- *    #endif
- *
-*/
-
-#define ROOT_RELEASE "6.29/01"
-#define ROOT_RELEASE_DATE "Dec 21 2022"
-#define ROOT_RELEASE_TIME "11:50:22"
-#define ROOT_VERSION(a,b,c) (((a) << 16) + ((b) << 8) + (c))
-#define ROOT_VERSION_CODE ROOT_VERSION(6,29,1) /* 400641 */
+#define ROOT_RELEASE_TIME "00:00:00" /* not updated anymore */
```
Comment (pcanal) [inline core/foundation/inc/RVersion.h:6]: Why is it no longer updated?  Should it be somehow marked as deprecated?

**Comment by pcanal** [comment]
The git log should be enhanced with the new 'procedure' for version update.

**Comment by pcanal** [comment]
One seeming drawback is that it is now 'possible' to forget to update the version number when going through the branching and the tagging for new versions.

**Review by pcanal — CHANGES_REQUESTED** [review]
See questions in the comments.

File: core/foundation/inc/RVersion.h (hunk 1 above)
Comment (Axel-Naumann) [reply core/foundation/inc/RVersion.h:6]: - I don't know how to mark it deprecated
- I don't think anyone cares about it not being "00:00:00"
- It's preprocessor stuff so I'd rather leave it alone :-)
-

**Comment by Axel-Naumann** [comment]
> > and- in its current implementation - a working ROOT build as part of the version update.

> Why is/was that?

Ouch that's wrong, that was true in the past, before I had replaced the ROOT script writing out the version number with a Python script. But the motivation for this PR is still there: instead of running a magic build step to update the version files we can now simply edit the header file.

**Comment by Axel-Naumann** [comment]
> The git log should be enhanced with the new 'procedure' for version update.

That seems to be a weird place, I'd rather want to keep it at https://root.cern/for_developers/release_checklist/

> One seeming drawback is that it is now 'possible' to forget to update the version number when going through the branching and the tagging for new versions.

How is that different than before, where `build/version_number` needed to be updated by hand?

**Comment by pcanal** [comment]
> and possibly at some point convertible to a Github action?

I am confused.  Isn't the new procedure `vi RVersion.hxx` and edit?

**Comment by Axel-Naumann** [comment]
> > and possibly at some point convertible to a Github action?
> I am confused. Isn't the new procedure vi RVersion.hxx and edit?

Yes. We could have a GH action that 
- does the edit, tags a release, and builds binaries, or
- tags the commit with a change to RVersion.hxx and builds the binaries, or
- reacts to the tag and does the build (my favorite).

All of that is possible today, it's just more involved because we need to run `make version` and then `cmake; make` again.

(In a next step I'll try to get rid of `gitinfo.txt` for releases; this is another extra complexity compared to just taking the sources from git.)

**Comment by pcanal** [comment]
> We could have a GH action that
> does the edit

So we still need the file `makeversion.py` just possibly with a different source of information, don't we?

**Comment by Axel-Naumann** [comment]
> So we still need the file makeversion.py just possibly with a different source of information, don't we?

Only if we take that route. In principle I want to have the sources edited and tagged, done. Straight forward :-)

**Review by pcanal — APPROVED** [review]
One of the remove will eventually be reintroduce but still good to go for now.

**Pull request merged**